### PR TITLE
Signup: Remove site mockup

### PR DESCRIPTION
**Pull request by arthur791004**
#### Changes proposed in this Pull Request

* As https://github.com/Automattic/wp-calypso/pull/59795 mentioned, the `onboarding-with-preview` flow is no longer in use. Also for the `<SiteMockups />` component. Thus, we can remove the `<SiteMockups />` component and site preview related code such `showSitePreview`, `hideSitePreview` and `isSitePreviewVisible`

#### Testing instructions

* Go to `/start`
* Ensure the flow works well after the removal

Related to https://github.com/Automattic/wp-calypso/issues/56264

### CODE REVIEW

**Comment by roo2** [comment]
Rebased onto trunk

**Review by roo2 — APPROVED** [review]
LGTM!


One small thing: a few places in `steps-pure.js` are still passing the `showSiteMockups` prop in

**Comment by arthur791004** [comment]
> One small thing: a few places in steps-pure.js are still passing the showSiteMockups prop in

Do you have a link to refer to one of those places as I don't see `showSiteMockups` anymore?

File: client/state/signup/preview/actions.js [hunk 1]
```diff
@@ -1,21 +0,0 @@
-import { SIGNUP_SITE_PREVIEW_SHOW, SIGNUP_SITE_PREVIEW_HIDE } from 'calypso/state/action-types';
```
Comment (tyxla) [inline client/state/signup/preview/actions.js:1]: You can also delete the action types now that they're not used.

File: client/signup/main.jsx [hunk 2]
```diff
@@ -68,7 +66,6 @@ import flows from './config/flows';
 import { getStepComponent } from './config/step-components';
 import steps from './config/steps';
 import { addP2SignupClassName } from './controller';
-import SiteMockups from './site-mockup';
```
Comment (tyxla) [inline client/signup/main.jsx:71]: We can also delete the `SiteMockups` component now that it's unused. And when we remove it, we can also remove:

* The `SignupSitePreview` component, its utils, and subcomponents
* The `client/lib/signup/site-theme` module

**Review by tyxla — COMMENTED** [review]
Thanks for the great cleanup! I think we can even delete some more in this PR:

File: client/signup/main.jsx (hunk 2 above)
Comment (arthur791004) [reply client/signup/main.jsx:71]: Oh yes, you're right! I've also removed site preview related data in our codebase 👍

**Comment by roo2** [comment]
> > One small thing: a few places in steps-pure.js are still passing the showSiteMockups prop in

> Do you have a link to refer to one of those places as I don't see showSiteMockups anymore?

Ahh sorry I retested and it's no longer there, perhaps my local env was out of date at the time 👍 

I found something else tiny when trying to work out what I was talking about. 

`shouldShowMockups` in https://github.com/Automattic/wp-calypso/blob/feat/remove-site-mockup/client/signup/main.jsx#L137 is no longer needed. It isn't actually directly part of this but it looks like it's been unused for a while and should have been taken out in [shouldShowMockups](https://github.com/Automattic/wp-calypso/commit/9d2ebe673c118b5d81c8734da7c47f16c70e4bdd)

**Pull request merged**